Selenium

orangebud opened the pull request:
Created PageObject classes, and now using  Selenium WebDriver to simulate user interactions for newsletter subscription.

orangebud committed:
Driver class created and added to selenium branch under the util package.

orangebud committed:
Page object classes added under the pageobjects package.

orangebud committed:
Refactored addRecord() method for use WebDriver to execute a subsrciption. After method added to close browser after the subscription is finished.

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java:
```diff
@@ -7,59 +7,77 @@
-	}
+    @Test
+    public void addRecord() throws Exception {
+        //given
```
Unnecessary to comment

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java:
```diff
@@ -7,59 +7,77 @@
-import org.junit.Test;
+    // Verification
+    private List<String> listToVerifyEmail;
+    private static final int NUMBER_OF_RESPONSE = 1;
```
I think this is a CS error. Constants should be defined before fields.
Run `mvn clean verify` before commit.

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/pageobjects/HomePageObject.java:
```diff
@@ -0,0 +1,110 @@
+         * @param newsLetterCheckbox Newsletter checkbox.
+         * @throws Exception When a field not visible or editable, or the page cannot be loaded.
+         */
+        public ThankYouPageObject givenSubcribeToNewsletter(String firstName, String lastName, String email,
```
Should be called signUp or something

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/util/Driver.java:
```diff
@@ -0,0 +1,40 @@
+ * Created by Peter_Olah1 on 12/10/2015.
+ */
+public class Driver {
+    private static WebDriver driver;
```
Generally its not good to have a shared static variable. This would kill parallel test execution because of the shared WD instance.

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/pageobjects/ThankYouPageObject.java:
```diff
@@ -0,0 +1,60 @@
+/**
+ * Created by Peter_Olah1 on 12/10/2015.
+ */
+public class ThankYouPageObject {
```
A PageObject, in general, should mimic a page's behaviour. So I do not prefer that validations are conducted by the PageObject.

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/pageobjects/HomePageObject.java:
```diff
@@ -0,0 +1,110 @@
+        @FindBy(id = "firstName")
+        private WebElement firstName;
+
+        @FindBy(css = "#lastName")
```
Could be id instead of css

orangebud committed:
Minor changes. Checkstyle errors fixed(still have some warning) and  most of the elements are searched by id locator instead of css.

orangebud committed:
Minor changes. Checkstyle errors fixed(still have some warning) and  most of the elements are searched by id locator instead of css.

orangebud committed:
Checkstyle errors fixed.

orangebud committed:
Minor changes. Method renamed as requested.

orangebud committed:
Minor changes. Checkstyle errors fixed.

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/pageobjects/HomePageObject.java:
```diff
@@ -0,0 +1,110 @@
+         * @param driver - {@link WebDriver}.
+         */
+        public HomePageObject(WebDriver driver) {
+            PageFactory.initElements(Driver.getDriver(), this);
```
You pass in a driver reference, why not use it?

orangebud committed:
- Class renamed
- Constructor parametrization modified as requested
- WebDriver is not static anymore

orangebud committed:
- New class, verifyer methods moved here from the page object class as requested

orangebud committed:
- New class, verifyer methods moved here from the page object class as requested

orangebud committed:
- Log4j dependency added

orangebud committed:
- Log4j config file added for proper working

orangebud committed:
-  Driver class is now abstract. Page object classes are extending it
- Constructor is modified

orangebud committed:
- Verifier methods are in separate class as requested
- Constructor modified

orangebud committed:
-  Logger configured
- Constructor modified
- After class modified

orangebud committed:
-  Class refactored to run parametrized tests with field injection

orangebud committed:
-  Logger used instead of sysout.
- Added javadoc comments.

orangebud committed:
-  Added javadoc comments

orangebud committed:
-  Unnecessary methods have been deleted

orangebud committed:
-  Acces modifier of the parameter variables changed to public

orangebud committed:
- A few web elements and getters added which were required by the new class

orangebud committed:
-  Minimal modification. throws declarations were removed

orangebud committed:
- 2 new class. Verifier contains home page verification methods, SubscriptionWithWebDriverTest responsible for automated test execution with junit

orangebud committed:
- WebDriver test execution class. Without rest.

tcsako commented on example-tests/pom.xml:
```diff
@@ -20,6 +20,11 @@
     <dependency>
+      <groupId>log4j</groupId>
+      <artifactId>log4j</artifactId>
+      <version>1.2.17</version>
```
please add the version into the parent pom.xml

tcsako commented on example-tests/log4j.xml:
```diff
@@ -0,0 +1,18 @@
+    <appender name="console" class="org.apache.log4j.ConsoleAppender">
+        <layout class="org.apache.log4j.PatternLayout">
+            <param name="ConversionPattern"
+                   value="%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n" />
```
please use US date format

tcsako commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java:
```diff
@@ -7,59 +7,104 @@
 //TODO: 5. create DDT script for REST script
```
remove those TODOs which are done

tcsako commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java:
```diff
@@ -7,59 +7,104 @@
+public class BasicServiceTest {
+    private static Logger log = Logger.getLogger(BasicServiceTest.class.getName());
+    private static final int NUMBER_OF_RESPONSE = 1;
+    private static final String CONTENT_NUMBER_OF_ELEMENTS = "numberOfElements";
```
remove these constants and use "numberOfElements", etc. in the rest verification

tcsako commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java:
```diff
@@ -7,59 +7,104 @@
+    private static final int NUMBER_OF_RESPONSE = 1;
+    private static final String CONTENT_NUMBER_OF_ELEMENTS = "numberOfElements";
+    private static final String CONTENT_EMAIL_ADDRESS = "content.emailAddress";
+    private static final int HTTP_OK = 200;
```
use existing enum for HTTP status code

tcsako commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java:
```diff
@@ -7,59 +7,104 @@
+        log.info("*******************************************");
+        log.info("Deleting existing records");
+        given().delete(ServiceTestingProperties.REST_API_URL);
 
```
log whether the delete was successful or not (e.g. verify status code)

tcsako commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java:
```diff
@@ -7,59 +7,104 @@
+        listToVerifyEmail = new ArrayList<String>();
+        listToVerifyEmail.add(emailAddress);
+        log.info("Initializing Firefox driver");
+        driver = new FirefoxDriver();
```
create abstract webdriver test class and define the before method there to initialize web driver object

tcsako commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java:
```diff
@@ -7,59 +7,104 @@
+    /**
+     *  Closes the browser after test execution.
+     */
+    @After
```
move this to the abstract class as well

tcsako commented on example-tests/src/test/java/com/epam/restassured/HomePageVerifier.java:
```diff
@@ -0,0 +1,70 @@
+     *
+     * @return HomePage instance.
+     */
+    public HomePageObject getInstance() {
```
do not use singleton pattern

tcsako commented on example-tests/src/test/java/com/epam/restassured/HomePageVerifier.java:
```diff
@@ -0,0 +1,70 @@
+     */
+    public HomePageObject getInstance() {
+        getDriver().get(BASE_URL);
+        HomePageObject homePage = PageFactory.initElements(getDriver(), HomePageObject.class);
```
move this to constructor

tcsako commented on example-tests/src/test/java/com/epam/restassured/SampleRestTest.java:
```diff
@@ -1,90 +1,146 @@
+     */
+    @Test
+    public void addRecord() {
+        List<String> list = new ArrayList<String>();
```
use meaningful name than 'list'

tcsako commented on example-tests/src/test/java/com/epam/restassured/SubscriptionWithWebDriverTest.java:
```diff
@@ -0,0 +1,81 @@
+ */
+public class SubscriptionWithWebDriverTest {
+
+    private Logger log = Logger.getLogger(BasicServiceTest.class.getName());
```
should be in the abstract test class

tcsako commented on example-tests/src/test/java/com/epam/restassured/ThankYouPageVerifier.java:
```diff
@@ -0,0 +1,57 @@
+ * Created by Peter_Olah1 on 12/10/2015.
+ */
+public class ThankYouPageVerifier extends AbstractDriver {
+    private ThankYouPageObject thanksPage = new ThankYouPageObject(getDriver());
```
delete

tcsako commented on example-tests/src/test/java/com/epam/restassured/pageobjects/HomePageObject.java:
```diff
@@ -0,0 +1,196 @@
+     *
+     * @return signUpForm
+     */
+    public WebElement getSignUpForm() {
```
remove all getters and use the webelements in public methods

tcsako commented on example-tests/src/test/java/com/epam/restassured/util/AbstractDriver.java:
```diff
@@ -0,0 +1,29 @@
+/**
+ * Created by Peter_Olah1 on 12/10/2015.
+ */
+public class AbstractDriver {
```
rename it to e.g. AbstractBasePage or something similar

orangebud committed:
- log4j dependency added to parent pom.xml

orangebud committed:
- log4j version removed from pom.xml(parent pom.xmml alredy contains it)

orangebud committed:
- http status is from org.apache.http.HttpStatus now

orangebud committed:
-  using US date format

orangebud committed:
- Received status is checked for 200 - ok
- Deletion success/faiurel is logged now

orangebud committed:
- Now using US date format

orangebud committed:
- BasePage renamed to AbstractBasePage

orangebud committed:
- class renamed to SignUpPageVerifier from HomePageVerifier
- moved to pageobjects package as requested
- constructor modified to avoid sinleton

orangebud committed:
- moved to pageobjects package as requested
- constructor modified to avoid singleton
- not extending the AbstractBasePage anymore

orangebud committed:
- minimal modifications( imports, logger and naming modifications)

orangebud committed:
-  clearing rubble after head hard resetting

orangebud committed:
- classes renamed(please check naming and function correctness)
- original class with REST added from the branch( logger added, now using HttpStatus instead of "200")
- BasicServiceTestWithWebDriver class is just performing a subscription on the UI and makes basic assertions between the expected and actual behavior

orangebud committed:
- unnecessary comment removed

orangebud committed:
Merge branch 'master' into selenium

Conflicts:
	example-tests/pom.xml
	example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java
	parent/pom.xml

orangebud committed:
Logger added, now using HttpStatus.SC_OK instead of "200" value

orangebud committed:
Now using the implemented CSV reader instead of local variables.

orangebud committed:
Minimal structural changes.

orangebud committed:
Minimal structural changes

orangebud committed:
Now using the implemented CSV reader instead of local variables. Minimal structural changes

orangebud committed:
Javadoc for getters removed. Unnecessary getters have been deleted

orangebud committed:
Unused variables removed. Javadoc comment added.

orangebud committed:
Unnecessary getter javadocs have been removed

orangebud committed:
Minimal changes. Unnecessary lines removed.

orangebud committed:
Data Driven Test implementation for SampleRestTest class

orangebud committed:
One line removed. Does not affects functionality

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTestWithWebDriver.java:
```diff
@@ -0,0 +1,103 @@
+    private CSVRestTestInput testInput;
+	// Default file name to read input data
+    private static final int HTTP_OK = HttpStatus.SC_OK;
+	private static final String DEFAULT_TEST_INPUT_FILE = "test_data_webdriver.csv";
```
Fix the indentation

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTestWithWebDriver.java:
```diff
@@ -0,0 +1,103 @@
+ */
+public class BasicServiceTestWithWebDriver {
+
+    private Logger log = Logger.getLogger(BasicServiceTestWithWebDriver.class);
```
Logger should be static filnal

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTest.java:
```diff
@@ -7,26 +7,28 @@
 	private static final String CONTENT_NUMBER_OF_ELEMENTS = "numberOfElements";
 	private static final String CONTENT_EMAIL_ADDRESS = "content.emailAddress";
-	private static final int HTTP_OK = 200;
+	private static final int HTTP_OK = HttpStatus.SC_OK;
```
It's unnecessary to create an other constant just to have an own reference

orangebud committed:
Indentation error fixed

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/BasicServiceTestWithRest.java:
```diff
@@ -0,0 +1,91 @@
+
+        log.info("*******************************************");
+        log.info("Deleting existing records");
+        if (given().delete(ServiceTestingProperties.REST_API_URL).getStatusCode() == httpOkStatus) {
```
This will always fail, you didn't initialized the field

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/SampleRestTest.java:
```diff
@@ -1,90 +1,76 @@
+        SubscriberResponse subscriberResponse = res.as(SubscriberResponse.class);
+        System.out.println(res.asString());
+        for (Content content : subscriberResponse.getContent()) {
+            System.out.println(content.getFirstName());
```
Use the logger

orangebud committed:
Loggers are static final now

jozsefkoza commented on example-tests/src/test/java/com/epam/restassured/SampleRestTest.java:
```diff
@@ -1,90 +1,76 @@
+    public void verifyResultNumber() {
+        Response res = get("https://t7-f0x.rhcloud.com/subscription/api/subscribers/?search=John");
+        JsonPath jp = new JsonPath(res.asString());
+        assertEquals("Result number should be 1", 1, jp.getInt("numberOfElements"));
```
You can get this directly through the response

orangebud committed:
Loggers are static final now

orangebud committed:
Loggers are static final now

orangebud committed:
Logger used instead of SysOut
